src/MoBi.Core/Services/AdjustFormulaVisitor.cs

```diff
@@ -1,30 +1,31 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using OSPSuite.Core.Commands.Core;
-using OSPSuite.Utility.Visitor;
 using MoBi.Core.Commands;
 using MoBi.Core.Domain.Model;
+using OSPSuite.Core.Commands.Core;
 using OSPSuite.Core.Domain;
 using OSPSuite.Core.Domain.Builder;
 using OSPSuite.Core.Domain.Formulas;
+using OSPSuite.Utility.Visitor;
 
 namespace MoBi.Core.Services
 {
    public interface IAdjustFormulasVisitor
    {
       /// <summary>
-      /// Adjusts the formula usage for the given objectBase in the buildingBlock.
-      /// The method looks if in the buildingBlocks formula cache the same formula (same name,paths,etc, not id)  is already used.
-      /// If it is, the already existing formula instance is used instead of the original one.
-      /// If the name is the same but other properties are different, the user is asked to rename the formula and then it's added to the formula cache
-      /// In other cases the new formula is added unchanged to the Formula cache
+      ///    Adjusts the formula usage for the given objectBase in the buildingBlock.
+      ///    The method looks if in the buildingBlocks formula cache the same formula (same name,paths,etc, not id)  is already
+      ///    used.
+      ///    If it is, the already existing formula instance is used instead of the original one.
+      ///    If the name is the same but other properties are different, the user is asked to rename the formula and then it's
+      ///    added to the formula cache
+      ///    In other cases the new formula is added unchanged to the Formula cache
       /// </summary>
       /// <param name="objectBase">The object base.</param>
       /// <param name="buildingBlock">The building block.</param>
       /// <returns></returns>
-      IEnumerable<IMoBiCommand> AdjustFormulasIn(IObjectBase objectBase, IBuildingBlock buildingBlock);
-      bool Canceled { get; }
+      (IReadOnlyList<IMoBiCommand> addedFormulaCommand, bool canceled) AdjustFormulasIn(IObjectBase objectBase, IBuildingBlock buildingBlock);
```

Comment:
simply use better (IMO) encapsulation by having the action return state and success